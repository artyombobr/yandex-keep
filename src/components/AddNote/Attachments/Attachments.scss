.attachments {
  display: flex;
  flex-direction: column;
}
