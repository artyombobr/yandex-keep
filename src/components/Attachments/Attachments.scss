.site {
  display: flex;
  align-items: center;
  margin: 0 24px;
  padding: 12px 0;
  border-bottom: 1px solid white;

  &:last-child {
    border: none;
  }

  &__link {
    overflow: hidden;
    text-overflow: ellipsis;
    margin-left: 12px;
    font-size: 16px;
    line-height: 24px;
    text-decoration: none;
    font-weight: bold;
    transition: color .4s;

    &:hover {
      color: #db5341;
    }
  }
}

.image {
  width: 64px;
  height: 64px;
  border-radius: 4px;
  margin: 12px 12px;
}

.images {
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin: 0 12px;

  &__icon {
    margin-left: 12px;
  }

  &__list {
    display: flex;
    flex-wrap: wrap;
    justify-content: flex-end;
  }
}
