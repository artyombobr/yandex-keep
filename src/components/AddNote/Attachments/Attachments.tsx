import React from 'react';
import './Attachments.scss';

const Attachments = (props: any) => {
  const { dataNote, setData } = props;
  return (
    <div className="attachments">
      {dataNote.attachments.map((attachment: any, index: number) => (
        <input
          className="attachments__input"
          placeholder="Введите ссылку"
          key={index.toString()}
          type="text"
          onChange={e => {
            const { value } = e.target;
            const { attachments } = dataNote;
            attachments[index] = { type: 'link', url: value };
            setData({ ...dataNote, attachments });
          }}
        />
      ))}
      <button
        type="button"
        className="modal__button attachments__button"
        onClick={() => {
          setData({ ...dataNote, attachments: [...dataNote.attachments, {}] });
        }}
      >
        {'Добавить'}
      </button>
    </div>
  );
};

export default Attachments;
