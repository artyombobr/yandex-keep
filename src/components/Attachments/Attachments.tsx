import React from 'react';
import linkCut from '../../helpers/linkCut';
import './Attachments.scss';

import { ReactComponent as Site } from '../Note/svg/site.svg';
import { ReactComponent as Picture } from '../Note/svg/picture.svg';

const Attachments = (props: any) => {
  const { attachments, isEdit } = props;
  const list = [];
  const listSites = [];
  const listImages = [];
  if (attachments !== undefined) {
    for (let i = 0; i < attachments.length; i++) {
      if (attachments[i].type === 'link') {
        listSites.push(
          <div className="site" key={i}>
            <Site />
            <a className="site__link" href={attachments[i].url}>
              {linkCut(attachments[i].url)}
            </a>
          </div>
        );
      } else {
        listImages.push(
          <img
            className="image"
            src={attachments[i].url}
            alt="Прикрепленное изображение"
            key={i}
          />
        );
      }
    }
  }
  listSites.length &&
    list.push(
      <div className="sites" key={0}>
        {listSites}
      </div>
    );
  listImages.length &&
    list.push(
      <div className="images" key={1}>
        <Picture className="images__icon" />
        <div className="images__list">{listImages}</div>
      </div>
    );
  return <>{list}</>;
};

export default Attachments;
