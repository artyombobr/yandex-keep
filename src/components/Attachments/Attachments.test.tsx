import React from 'react';
import { shallow } from 'enzyme';
import Attachments from './Attachments';

describe('Attachments', () => {
  const attachments = [
    { type: 'link', url: 'https://yandex.ru' },
    {
      type: 'image',
      url:
        'https://avatars.mds.yandex.net/get-pdb/1816426/93eab951-b130-4cf9-98d6-01e250be5530/orig',
    },
  ];
  it('renders correctly', () => {
    const component = shallow(<Attachments attachments={attachments} />);
    expect(component).toMatchSnapshot();
  });
});
