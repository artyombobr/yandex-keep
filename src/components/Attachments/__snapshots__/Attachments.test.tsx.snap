// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Attachments correct render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Attachments
    attachments={
      Array [
        Object {
          "type": "link",
          "url": "https://yandex.ru",
        },
        Object {
          "type": "image",
          "url": "https://avatars.mds.yandex.net/get-pdb/1816426/93eab951-b130-4cf9-98d6-01e250be5530/orig",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          className="sites"
        >
          <div
            className="site"
          >
            <ForwardRef />
            <a
              className="site__link"
              href="https://yandex.ru"
            >
              yandex.ru
            </a>
          </div>
        </div>,
        <div
          className="images"
        >
          <ForwardRef
            className="images__icon"
          />
          <div
            className="images__list"
          >
            <img
              alt="Прикрепленное изображение"
              className="image"
              src="https://avatars.mds.yandex.net/get-pdb/1816426/93eab951-b130-4cf9-98d6-01e250be5530/orig"
            />
          </div>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="site"
            >
              <ForwardRef />
              <a
                className="site__link"
                href="https://yandex.ru"
              >
                yandex.ru
              </a>
            </div>,
          ],
          "className": "sites",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <ForwardRef />,
                <a
                  className="site__link"
                  href="https://yandex.ru"
                >
                  yandex.ru
                </a>,
              ],
              "className": "site",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                },
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "yandex.ru",
                  "className": "site__link",
                  "href": "https://yandex.ru",
                },
                "ref": null,
                "rendered": "yandex.ru",
                "type": "a",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ForwardRef
              className="images__icon"
            />,
            <div
              className="images__list"
            >
              <img
                alt="Прикрепленное изображение"
                className="image"
                src="https://avatars.mds.yandex.net/get-pdb/1816426/93eab951-b130-4cf9-98d6-01e250be5530/orig"
              />
            </div>,
          ],
          "className": "images",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "images__icon",
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt="Прикрепленное изображение"
                  className="image"
                  src="https://avatars.mds.yandex.net/get-pdb/1816426/93eab951-b130-4cf9-98d6-01e250be5530/orig"
                />,
              ],
              "className": "images__list",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "host",
                "props": Object {
                  "alt": "Прикрепленное изображение",
                  "className": "image",
                  "src": "https://avatars.mds.yandex.net/get-pdb/1816426/93eab951-b130-4cf9-98d6-01e250be5530/orig",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="sites"
          >
            <div
              className="site"
            >
              <ForwardRef />
              <a
                className="site__link"
                href="https://yandex.ru"
              >
                yandex.ru
              </a>
            </div>
          </div>,
          <div
            className="images"
          >
            <ForwardRef
              className="images__icon"
            />
            <div
              className="images__list"
            >
              <img
                alt="Прикрепленное изображение"
                className="image"
                src="https://avatars.mds.yandex.net/get-pdb/1816426/93eab951-b130-4cf9-98d6-01e250be5530/orig"
              />
            </div>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="site"
              >
                <ForwardRef />
                <a
                  className="site__link"
                  href="https://yandex.ru"
                >
                  yandex.ru
                </a>
              </div>,
            ],
            "className": "sites",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <ForwardRef />,
                  <a
                    className="site__link"
                    href="https://yandex.ru"
                  >
                    yandex.ru
                  </a>,
                ],
                "className": "site",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "render": [Function],
                  },
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "yandex.ru",
                    "className": "site__link",
                    "href": "https://yandex.ru",
                  },
                  "ref": null,
                  "rendered": "yandex.ru",
                  "type": "a",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <ForwardRef
                className="images__icon"
              />,
              <div
                className="images__list"
              >
                <img
                  alt="Прикрепленное изображение"
                  className="image"
                  src="https://avatars.mds.yandex.net/get-pdb/1816426/93eab951-b130-4cf9-98d6-01e250be5530/orig"
                />
              </div>,
            ],
            "className": "images",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "images__icon",
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <img
                    alt="Прикрепленное изображение"
                    className="image"
                    src="https://avatars.mds.yandex.net/get-pdb/1816426/93eab951-b130-4cf9-98d6-01e250be5530/orig"
                  />,
                ],
                "className": "images__list",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "1",
                  "nodeType": "host",
                  "props": Object {
                    "alt": "Прикрепленное изображение",
                    "className": "image",
                    "src": "https://avatars.mds.yandex.net/get-pdb/1816426/93eab951-b130-4cf9-98d6-01e250be5530/orig",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
